frontend/javascripts/oxalis/model/reducers/volumetracing_reducer.ts

```diff
@@ -194,6 +198,37 @@ function handleUpdateSegment(state: OxalisState, action: UpdateSegmentAction) {
   });
 }
 
+function handleFocusSegment(state: OxalisState, action: ClickSegmentAction) {
+  if (action.layerName == null) return state;
+  const getNewGroups = () => {
+    const { segments, segmentGroups } = getVisibleSegments(state);
+    if (action.layerName == null || segments == null) return segmentGroups;
+    const { segmentId } = action;
+    const segmentForId = segments.getNullable(segmentId);
+    if (segmentForId == null) return segmentGroups;
+    // Expand recursive parents of group too, if necessary
+    let pathToRoot = [];
+    if (segmentForId.groupId != null) {
+      let currentParent = findParentIdForGroupId(segmentGroups, segmentForId.groupId);
+      while (currentParent != null) {
+        pathToRoot.push(currentParent);
+        currentParent = findParentIdForGroupId(segmentGroups, currentParent);
+      }
+    }
+    const pathToRootSet = new Set(pathToRoot);
+    return mapGroups(segmentGroups, (group) => {
+      if (
+        (pathToRootSet.has(group.groupId) || group.groupId === segmentForId.groupId) &&
```

Comment:
how about initializing pathToRootSet so that `segmentForId.groupId` is already included. then, you only need the `has` check here.